x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/action/role/PutRoleRequestTests.java

```diff
@@ -185,8 +186,32 @@ private PutRoleRequest buildRandomRequest() {
         request.addApplicationPrivileges(applicationPrivileges);
 
         if (randomBoolean()) {
-            final String[] appNames = randomArray(1, 4, String[]::new, stringWithInitialLowercase);
-            request.conditionalCluster(new ConfigurableClusterPrivileges.ManageApplicationPrivileges(Sets.newHashSet(appNames)));
+            if (randomBoolean()) {
+                request.conditionalCluster(new ConfigurableClusterPrivilege[0]);
+            } else {
+                request.conditionalCluster(
+                    new ConfigurableClusterPrivileges.ManageApplicationPrivileges(
+                        Sets.newHashSet(randomArray(0, 3, String[]::new, stringWithInitialLowercase))
```

Comment:
> We don't alow empty list of index names. So why should we allow empty list of application names. There is no real use case for it.

I think it can be convenient to support an empty array if the role descriptor is generated programatically.

> I struggle to justify that we should actively assert/test this behaviour since it might give people the impression that we conciously support it.

I'm on the opposite side: if the main code supports it we should test for it, even though it is not an interesting test.

Reply:
> I think it can be convenient to support an empty array if the role descriptor is generated programatically.

I have a different view on this usage. If the progamatical generation process, e.g. Mustache templates, has a bug and leads to empty array, we will not be able to catch it if we accept empty arrays. I'd think it is more of an issue than worth to allow empty arrays. But since this is almost the existing behaviour, I won't dwell on my argument.